### PR TITLE
Server related page updates

### PR DESCRIPTION
**Pull request by tib**
- alter server & sswg pages
- hide doc links from server documentation page 
- new sswg incubation-process page

### CODE REVIEW

**Review by ktoso — APPROVED** [review]
Good update, thank you!

File: documentation/server/index.md [hunk 1]
```diff
@@ -59,30 +59,6 @@ The Swift ecosystem contains many useful libraries and tools specifically design
 Overall, Swift on Server opens up new opportunities for developers to build fast, scalable, and secure backend services. Swift's combination of performance, readability, interoperability, safety, and modern language features make it a compelling choice for many developers.
 
 
-## Development guides
```
Comment (0xTim) [inline documentation/server/index.md:62]: Why did we decide to hide all these? They won't be linked to anywhere which I remember some discussion about removing some of them but things like debugging memory leaks and log levels are still useful

Comment (tib) [reply documentation/server/index.md:62]: I believe so, but I can put a link with a sort description to the server page, under the Supported ecosystem section, which leads to the development guides page, if needed, e.g.:

```
### Development guides

The Swift Server Workgroup and Swift on Server community have developed a number of [guides](/documentation/server/guides/) for using Swift on the server. They are designed to help teams and individuals running Swift Server applications on Linux and to provide orientation for those who want to start with such development.
``` 

Should I add back this to keep the link between the two pages?

Comment (0xTim) [reply documentation/server/index.md:62]: Yeah I think adding a link would be a good stopgap 👍

Comment (tib) [reply documentation/server/index.md:62]: Done.

**Review by 0xTim — APPROVED** [review]

**Comment by 0xTim** [comment]
@swift-ci test

**Pull request merged**